python/guide/modules/autograd.md

```diff
@@ -1,178 +1,260 @@
+
+Automatic Differentiation creates a record of the operators used (i.e. the `forward` method calls) by the network to make predictions and calculate the loss metric. A graph structure is used to record this, capturing the inputs (including their value) and outputs for each operator and how the operators are related. We call this the 'forward pass' of training.
+
+Automatic Differentiation then works backwards through each operator of the graph, calling the `backward` method on each operator to calculate the partial derivative and calculate the gradient of the loss metric with respect to the operator's input (which could be parameters). Usually we work backwards from the loss metric, and hence calculate the gradients of the loss metric, but this can be done from any output. We call this the 'backward pass' of training.
+
+### What are the advantages of Automatic Differentiation (`autograd`)?
+
+It's flexible, automatic and efficient. You can use [native Python control flow operators](#controlflow) such as `if` conditions and `while` loops and `autograd` will still be able to backpropogate the gradients correctly.
+
+Long Answer:
+
+A huge benefit of using `autograd` is the flexibility it gives you when defining your network. You can change the operations on every iteration, and `autograd` will still be able to backpropagate the gradients correctly. You'll sometimes hear these called 'dynamic graphs', and are much more complex to implement in frameworks that require static graphs, such as TensorFlow.
+
+As suggested by the name, `autograd` is automatic and so the complexities of the backpropagation procedure are taken care of for you. All you have to do is tell `autograd` when you're [interested in recording gradients](#nonparamgrad), and specify what gradients you're interested in calculating: this will nearly always just be the gradient of the loss metric. And these gradient calculations will be performed efficiently too.
+
+## How do I use `autograd` in MXNet Gluon?
+
+Step one is to import the `autograd` package.
 
 ```{.python .input  n=1}
-from mxnet import autograd, nd
+from mxnet import autograd
 ```
 
-## A Simple Example
-
-As a toy example, let's say that we are interested in
-differentiating the mapping $y = 2\mathbf{x}^{\top}\mathbf{x}$ with respect to
-the column vector $\mathbf{x}$. Firstly, we create the variable `x` and assign
-an initial value.
+As a simple example, we'll implement the regression model shown in the diagrams above, and later use `autograd` to automatically calculate the gradient of the loss with respect to each of the weight parameters.
 
 ```{.python .input  n=2}
-x = nd.arange(4).reshape((4, 1))
-print(x)
+from mxnet.gluon.nn import HybridSequential, Dense
+from mxnet.gluon.loss import L2Loss
+from mxnet.ndarray import array
+
+# define network
+net = HybridSequential()
+net.add(Dense(units=3))
+net.add(Dense(units=1))
```

Comment:
you could combine the two Dense's into one line

Reply:
would prefer to seperate the layers here.